design-proposals/app-orch-dp-import.md

```diff
@@ -0,0 +1,121 @@
+# Design Proposal: Onboarding Deployment Packages from Github Repositories
+
+Author(s): Scott Baker
+
+Last updated: 2025-04-16
+
+## Abstract
+
+This proposal describes a mechanism for importing a deployment package from an opensource
+application repository into the Orchestrator.
+
+## Proposal
+
+The convention chosen for opensource applications wishing to distribute their deployment packages
+is that deployment packages will be placed in yaml format into well-known `deployment-package/` directories. This
+convention has already been adopted for AI Suite applications at Intel, and endorsed as part of the
+Edge Software Catalog (ESC).
+
+Current mechanisem in 3.0 requires a user to clone the application repository and then import the deployment
+package into the Orchestrator using their web browser. This proposal seeks to eliminate the need
+to clone and import.
+
+This proposal is complicated by the fact that many repositories contain multiple deployment
+packages, and they are often located in subdirectories, and not all AI Suite repositories use the
+same subdirectory naming convention.
+
+There are two variants of this proposal:
+
+### Sub-Proposal - Import a specific deployment package from a repository (requires more GUI work)
+
+- The existing `Import Deployment Package` screen shall be updated with an `Import from URL` button.
+
+- The user will paste the URL of the repo they wish to import into the GUI.
+
+- The GUI will pass the URL to the application catalog service. For example, `/v3/projects/{project-name}/catalog/list_packages_from_repo?url=<url>`.
+
+- The application catalog shall scan the repository and return a list of deployment packages that were found in the
+  repository.
+
+- The GUI will display the list of deployment packages that were found and allow the user to choose one.
```

Comment:
This operation we should not expect the App Catalog service to cache or store information of the github repo and the deployment packages present in them.  We need to ensure these operations are stateless.

Reply:
clarified.